src/commcare_cloud/fab/operations/release.py

```diff
@@ -304,6 +301,12 @@ def update():
         # Optimization if we have current setup (i.e. not the first deploy)
         if exists(env.py3_virtualenv_current) and not exists(env.py3_virtualenv_root):
             _clone_virtual_env(env.py3_virtualenv_current, env.py3_virtualenv_root)
+        elif not exists(env.py3_virtualenv_current):
+            raise EnvironmentException(
+                "{} cannot be found. This means there is no virtual environment available to clone from. "
+                "You will need to recreate the /current directory on this machine that symlinks to the most " 
+                "recent release in order to deploy successfully.".format(env.py3_virtualenv_current)
```

Comment:
You mean does string literal concatenation happen before formatting? Yes
```py
>>> "x{}"  "z".format("y")
'xyz'
```

However, missing the `current` symlink is just one possible reason this condition might fail, so the error message seems a bit more specific than necessary.

```suggestion
                "Virtual environment not found: {}".format(env.py3_virtualenv_current)
```